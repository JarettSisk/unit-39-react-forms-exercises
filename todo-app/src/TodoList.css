.TodoList {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.TodoList-title {
  color: rgb(0, 247, 255);
  margin: 10px;
}