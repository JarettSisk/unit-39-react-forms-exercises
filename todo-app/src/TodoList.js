import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
import NewTodoForm from "./NewTodoForm";
import Todo from "./Todo";
import "./TodoList.css";

const TodoList = () => {
  const INITIAL_STATE = [
    {
      description: "wash dishes",
      id: uuid()
    },
    {
      description: "Do laundry",
      id: uuid()
    }
  ];

  const [todos, setTodos] = useState(INITIAL_STATE);

  const addTodo = (newTodo) => {
    setTodos(todos => [...todos, {...newTodo, id: uuid()}]);
  }

    const remove = (todoId) => {
      // set boxes to new array only containing boxes that dont match the id of the one we want to remove
      setTodos(todos.filter(t => t.id !== todoId));
  };

  const renderTodos = (todos) => {
    return todos.map((todo) => {
      return <Todo
        key={todo.id}
        description={todo.description}
        id={todo.id} 
        remove={remove}
        />
      });
  } 

  return (
    <div className="TodoList">
      <h1 className="TodoList-title">Todo List</h1>
      <NewTodoForm addTodo={addTodo}/>
      {renderTodos(todos)}
    </div>
  )
}

export default TodoList;