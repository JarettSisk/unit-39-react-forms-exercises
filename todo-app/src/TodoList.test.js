import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TodoList from "./TodoList";

// Smoke test
it("renders without crashing", function() {
    render(<TodoList />);
});
// Snapshot test
it("matches snapshot", () => {
    const { asFragment } = render(<TodoList />);
    expect(asFragment()).toMatchSnapshot();
});

it("should add a new Todo", () => {
  const {queryByLabelText, queryByText} = render(<TodoList />);
  const todoInput = queryByLabelText("What do you need to do?");
  const submitNewTodoButton = queryByText("ADD");

  expect(queryByText("fetch groceries")).not.toBeInTheDocument;

  fireEvent.change(todoInput, {target: {value: "fetch groceries"}});
  fireEvent.click(submitNewTodoButton);

  expect(queryByText("fetch groceries")).not.toBeInTheDocument;
})