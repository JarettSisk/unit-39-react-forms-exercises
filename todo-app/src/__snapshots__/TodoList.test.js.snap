// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="TodoList"
  >
    <h1
      class="TodoList-title"
    >
      Todo List
    </h1>
    <form
      class="NewTodoForm"
    >
      <label
        class="NewTodoFOrm-label-description"
        for="description"
      >
        What do you need to do?
      </label>
      <input
        class="NewTodoForm-input-description"
        id="description"
        name="description"
        placeholder="Pick up groceries"
        type="text"
        value=""
      />
      <button
        class="NewTodoForm-button"
      >
        ADD
      </button>
    </form>
    <div
      class="Todo"
    >
      <p
        class="Todo-description"
      >
        wash dishes
      </p>
      <button
        class="Todo-button"
      >
        X
      </button>
    </div>
    <div
      class="Todo"
    >
      <p
        class="Todo-description"
      >
        Do laundry
      </p>
      <button
        class="Todo-button"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
